src/ui/map.js

```diff
@@ -1056,19 +1056,25 @@ class Map extends Camera {
      * Returns the map's language, which is used for translating map labels and UI components.
      *
      * @private
-     * @returns {string} Returns the map's language code.
+     * @returns {undefined | string | string[]} Returns the map's language code.
      * @example
      * const language = map.getLanguage();
      */
-    getLanguage(): ?string {
+    getLanguage(): ?string | ?string[] {
         return this._language;
     }
 
+    _parseLanguage(language?: 'auto' | ?string | ?string[]): ?string | ?string[] {
+        if (language === 'auto') return window.navigator.language;
+        if (Array.isArray(language)) return language.length > 0 ? language : undefined;
```

Comment:
How is `undefined` treated down the line? Should we treat `null`/`undefined`/`[]` the same as `auto` here?

Reply:
@mourner `null`/`undefined`/`[]` are treated as an "empty" value that doesn't pass the language down to the vector tile requests. `auto` is the opposite — it always passes down the user's preferred language.

At the same time, it might be useful to find and replace an `auto` value if the array of strings is passed as an argument, what do you think?